packages/server/tests/helpers/utils.ts

```diff
@@ -372,4 +383,229 @@ export class Utils {
+      writeSnapshot((error, fileName) => {
+        if (error) {
+          reject(error);
+        }
+        console.info(`Heap snapshot written to ${fileName}`);
+        resolve(fileName);
+      });
+    });
+  }
+
+  /**
+   * Captures memory leaks in the test suite.
+   * The function will start the profiler before each test and stop it after each test.
+   * If a memory leak is detected, the function will log the difference in memory usage.
+   */
+  static captureMemoryLeaks(profiler: GCProfiler): void {
+    setFlagsFromString('--expose_gc');
+    const gc = runInNewContext('gc');
+    const githubClient = new GitHubClient();
+
+    let isWarmUpCompleted = false;
+
+    const warmUp = async () => {
+      for (let i = 0; i < Utils.WARM_UP_TEST_COUNT; i++) {
+        // Run dummy tests to warm up the environment
+        await new Promise((resolve) => setTimeout(resolve, 100));
+      }
+      isWarmUpCompleted = true;
+    };
+
+    beforeEach(async function () {
+      if (!isWarmUpCompleted) {
+        await warmUp();
+      }
+      profiler.start();
+    });
+
+    afterEach(async function (this: Context) {
+      this.timeout(60000);
+      await gc(); // force a garbage collection to get accurate memory usage
+      try {
+        const result = profiler.stop();
+        const statsGrowingHeapSize = result.statistics.filter((stats) => {
+          return stats.afterGC.heapStatistics.totalHeapSize > stats.beforeGC.heapStatistics.totalHeapSize;
+        });
+        const totalDiffBytes = statsGrowingHeapSize.reduce((acc, stats) => {
+          const diff = stats.afterGC.heapStatistics.totalHeapSize - stats.beforeGC.heapStatistics.totalHeapSize;
+          return acc + diff;
+        }, 0);
+        const isPotentialMemoryLeak = totalDiffBytes > Utils.HEAP_SIZE_DIFF_MEMORY_LEAK_THRESHOLD;
+
+        if (isPotentialMemoryLeak) {
+          console.warn('Potential memory leak detected!');
+          const statsDiff: HeapDifferenceStatistics = statsGrowingHeapSize.map((stats) => ({
+            gcType: stats.gcType,
+            cost: stats.cost,
+            diffGC: {
+              heapStatistics: Utils.difference(stats.afterGC.heapStatistics, stats.beforeGC.heapStatistics),
+              heapSpaceStatistics: Utils.difference(
+                stats.afterGC.heapSpaceStatistics,
+                stats.beforeGC.heapSpaceStatistics,
+              ).filter((spaceStatistics) => spaceStatistics.spaceSize > 0),
+            },
+          }));
+          console.error(
+            `Total Heap Size ${Utils.formatBytes(totalDiffBytes)}: --> ` + JSON.stringify(statsDiff, null, 2),
+          );
+          // add comment on PR highlighting after which test the memory leak is happening
+          const testTitle = this.currentTest?.title ?? 'Unknown test';
+          const comment = Utils.generateMemoryLeakComment(testTitle, statsDiff);
+          await githubClient.addOrUpdateExistingCommentOnPullRequest(comment, (existing: string) =>
```

Comment:
The addOrUpdateExistingCommentOnPullRequest() will first pull the context from GtiHub first for stuff like owner, repo, pullNumber, etc. right? Then just curious how this will play out locally when no GitHub-related-env-vars are provided?

Reply:
It wouldn’t work locally, but it will only log an error, tests will still pass